cloud_guardian/iam_model/graph/permission/actions.py

```diff
@@ -102,13 +139,34 @@ class ActionFactory:
         PutObject.aws_action: PutObject,
         Any.aws_action: Any,
         AssumeRole.aws_action: AssumeRole,
+        # NOTE: should all actions be defined and listed in the code?
+        DevS3Access.aws_action: DevS3Access,
+        ReadOnly.aws_action: ReadOnly,
+        Admin.aws_action: Admin,
     }
 
+    def get_action(cls, aws_action: str) -> SupportedAction:
+        if aws_action in cls._action_types:
+            return cls._action_types[aws_action]
+        else:
+            # TODO: improve the implementation to make it
+            # more robust (the idea is to identify the relevant
+            # action when the `_action_types` object does not
+            # strictly contains the aws_action name)
+            for k, v in cls._action_types.items():
+                if re.search(aws_action, k, re.IGNORECASE):
+                    return v
+
+        raise ActionNotSupported(aws_action)
+
     @classmethod
-    def get_or_create(cls, aws_action: str):
+    def get_or_create(cls, aws_action: str) -> SupportedAction:
+        aws_action = extract_policy_from_ARN(aws_action)
         if aws_action not in cls._instances:
-            if aws_action in cls._action_types:
-                action_class = cls._action_types[aws_action]
+            # TODO: should be refactor to properly support wildcards
```

Comment:
Yes, we should do something with the wildcards.
We should map a set of actions (with/without wildcards) to the supported actions, and ignore the rest if we don't have a match